packages/salesforcedx-vscode-core/src/commands/functions/forceFunctionStart.ts

```diff
@@ -5,40 +5,150 @@
+  /**
+   * Locate the directory that has function.toml
+   * @param sourceFsPath path to start function from
+   */
+  public static getFunctionDir(sourceFsPath: string) {
+    let current = fs.lstatSync(sourceFsPath).isDirectory()
+      ? sourceFsPath
+      : path.dirname(sourceFsPath);
+    const { root } = path.parse(sourceFsPath);
+    while (current !== root) {
+      const tomlPath = path.join(current, 'function.toml');
+      if (fs.existsSync(tomlPath)) {
+        return current;
+      }
+      current = path.dirname(current);
+    }
+    return undefined;
+  }
+
+  public execute(response: ContinueResponse<string>) {
+    const startTime = process.hrtime();
+    const cancellationTokenSource = new vscode.CancellationTokenSource();
+    const cancellationToken = cancellationTokenSource.token;
+    const sourceFsPath = response.data;
+    const functionDirPath = ForceFunctionStartExecutor.getFunctionDir(
+      sourceFsPath
+    );
+    if (!functionDirPath) {
+      notificationService.showWarningMessage(
+        nls.localize('force_function_start_warning_no_toml')
+      );
+      telemetryService.sendException(
+        'force_function_start',
+        'force_function_start_no_toml'
+      );
+      return;
+    }
+    const execution = new CliCommandExecutor(this.build(functionDirPath), {
+      cwd: this.executionCwd,
+      env: { SFDX_JSON_TO_STDOUT: 'true' }
+    }).execute(cancellationToken);
+    const executionName = execution.command.toString();
+
+    FunctionService.instance.registerStartedFunction({
+      terminate: () => {
+        return execution.killExecution('SIGTERM');
+      }
+    });
+
+    channelService.streamCommandOutputWithoutColor(execution);
+    channelService.showChannelOutput();
+
+    const progress = new Subject();
+    ProgressNotification.show(
+      execution,
+      cancellationTokenSource,
+      vscode.ProgressLocation.Notification,
+      progress.asObservable() as Observable<number>
+    );
+    const task = taskViewService.addCommandExecution(
+      execution,
+      cancellationTokenSource
+    );
+
+    execution.stdoutSubject.subscribe(data => {
+      if (data.toString().includes('Ready to process signals')) {
+        progress.complete();
+        taskViewService.removeTask(task);
+        notificationService
+          .showSuccessfulExecution(executionName)
+          .catch(() => {});
+        this.logMetric(execution.command.logName, startTime);
+      }
+    });
+
+    execution.processExitSubject.subscribe(async exitCode => {});
+
+    notificationService.reportCommandExecutionStatus(
+      execution,
+      cancellationToken
+    );
+  }
 }
 
 /**
  * Executes sfdx evergreen:function:start --verbose
  * @param sourceUri
  */
-export async function forceFunctionStart(sourceUri: Uri) {
+export async function forceFunctionStart(sourceUri?: Uri) {
+  if (!sourceUri) {
+    // Try to start function from current active editor, if running SFDX: start function from command palette
```

Comment:
We'll start function from the current active text editor if running "SFDX: Start Function" from command palette. We could potentially add a picker for multiple functions later

Reply:
You can also start function from right click in a file (for example `index.ts`), or right click in the function folder or file in explorer view